File: bindings/typescript/examples/basic-usage.ts
/**
 * Basic usage example for Satox SDK TypeScript Bindings
 */

import {
    CoreManager,
    DatabaseManager,
    BlockchainManager,
    AssetManager,
    WalletManager,
    DatabaseType,
    NetworkType,
    CoreConfig,
    DatabaseConfig,
    SatoxError
} from '../src';

async function main() {
    console.log('🚀 Satox SDK TypeScript Example');
    console.log('================================');
    
    // Initialize core manager
    console.log('\n1. Initializing Core Manager...');
    const coreManager = new CoreManager();
    const config: CoreConfig = {
        name: 'satox_sdk_example',
        enableLogging: true,
        logLevel: 'info',
        maxThreads: 4
    };
    
    try {
        await coreManager.initialize(config);
        console.log('✅ Core Manager initialized successfully');
        
        // Get status
        const status = await coreManager.getStatus();
        console.log('📊 SDK Status:', status);
        
    } catch (error) {
        console.error('❌ Failed to initialize Core Manager:', error);
        return;
    }
    
    // Initialize database manager
    console.log('\n2. Initializing Database Manager...');
    const dbManager = new DatabaseManager();
    const dbConfig: DatabaseConfig = {
        type: DatabaseType.SQLITE,
        connectionString: ':memory:'
    };
    
    try {
        await dbManager.initialize(dbConfig);
        await dbManager.connect(DatabaseType.SQLITE, ':memory:');
        console.log('✅ Database Manager initialized and connected');
        
        // Execute a test query
        const result = await dbManager.executeQuery('SELECT 1 as test');
        console.log('📊 Test query result:', result);
        
    } catch (error) {
        console.warn('⚠️  Database Manager issue:', error);
    }
    
    // Initialize blockchain manager
    console.log('\n3. Initializing Blockchain Manager...');
    const blockchainManager = new BlockchainManager();
    
    try {
        await blockchainManager.initialize({});
        await blockchainManager.connect(
            NetworkType.MAINNET,
            'http://localhost:7777',
            { username: 'user', password: 'pass' }
        );
        console.log('✅ Blockchain Manager initialized and connected');
        
        // Get block height
        const height = await blockchainManager.getBlockHeight();
        console.log('📊 Current block height:', height);
        
    } catch (error) {
        console.warn('⚠️  Blockchain Manager issue:', error);
    }
    
    // Initialize asset manager
    console.log('\n4. Initializing Asset Manager...');
    const assetManager = new AssetManager();
    
    try {
        await assetManager.initialize({});
        console.log('✅ Asset Manager initialized');
        
        // Create a test asset
        const assetConfig = {
            name: 'Test Token',
            symbol: 'TEST',
            description: 'A test token created with Satox SDK',
            precision: 8,
            reissuable: true,
            totalSupply: 1000000
        };
        
        const asset = await assetManager.createAsset(assetConfig);
        console.log('✅ Asset created:', asset.name, `(${asset.symbol})`);
        
    } catch (error) {
        console.warn('⚠️  Asset Manager issue:', error);
    }
    
    // Initialize wallet manager
    console.log('\n5. Initializing Wallet Manager...');
    const walletManager = new WalletManager();
    
    try {
        await walletManager.initialize({});
        console.log('✅ Wallet Manager initialized');
        
        // Create a wallet
        const wallet = await walletManager.createWallet();
        console.log('✅ Wallet created:', wallet.address);
        
    } catch (error) {
        console.warn('⚠️  Wallet Manager issue:', error);
    }
    
    // Shutdown
    console.log('\n6. Shutting down...');
    try {
        await coreManager.shutdown();
        console.log('✅ SDK shutdown successfully');
    } catch (error) {
        console.warn('⚠️  Shutdown issue:', error);
    }
    
    console.log('\n🎉 Example completed successfully!');
}

// Run the example
if (require.main === module) {
    main().catch(console.error);
} 

File: bindings/python/examples/basic_usage.py
"""
Basic usage example for Satox SDK Python Bindings
"""

import asyncio
from satox_bindings import (
    CoreManager,
    DatabaseManager,
    BlockchainManager,
    AssetManager,
    WalletManager,
    DatabaseType,
    NetworkType,
    CoreConfig,
    DatabaseConfig,
    SatoxError
)

async def main():
    print("🚀 Satox SDK Python Example")
    print("============================")
    
    # Initialize core manager
    print("\n1. Initializing Core Manager...")
    core_manager = CoreManager()
    config = CoreConfig(
        name="satox_sdk_example",
        enable_logging=True,
        log_level="info",
        max_threads=4
    )
    
    try:
        await core_manager.initialize(config)
        print("✅ Core Manager initialized successfully")
        
        # Get status
        status = await core_manager.get_status()
        print(f"📊 SDK Status: {status}")
        
    except SatoxError as e:
        print(f"❌ Failed to initialize Core Manager: {e}")
        return
    
    # Initialize database manager
    print("\n2. Initializing Database Manager...")
    db_manager = DatabaseManager()
    db_config = DatabaseConfig(
        type=DatabaseType.SQLITE,
        connection_string=":memory:"
    )
    
    try:
        await db_manager.initialize(db_config)
        await db_manager.connect(DatabaseType.SQLITE, ":memory:")
        print("✅ Database Manager initialized and connected")
        
        # Execute a test query
        result = await db_manager.execute_query("SELECT 1 as test")
        print(f"📊 Test query result: {result}")
        
    except SatoxError as e:
        print(f"⚠️  Database Manager issue: {e}")
    
    # Initialize blockchain manager
    print("\n3. Initializing Blockchain Manager...")
    blockchain_manager = BlockchainManager()
    
    try:
        await blockchain_manager.initialize({})
        await blockchain_manager.connect(
            NetworkType.MAINNET,
            "http://localhost:7777",
            {"username": "user", "password": "pass"}
        )
        print("✅ Blockchain Manager initialized and connected")
        
        # Get block height
        height = await blockchain_manager.get_block_height()
        print(f"📊 Current block height: {height}")
        
    except SatoxError as e:
        print(f"⚠️  Blockchain Manager issue: {e}")
    
    # Initialize asset manager
    print("\n4. Initializing Asset Manager...")
    asset_manager = AssetManager()
    
    try:
        await asset_manager.initialize({})
        print("✅ Asset Manager initialized")
        
        # Create a test asset
        asset_config = {
            "name": "Test Token",
            "symbol": "TEST",
            "description": "A test token created with Satox SDK",
            "precision": 8,
            "reissuable": True,
            "total_supply": 1000000
        }
        
        asset = await asset_manager.create_asset(asset_config)
        print(f"✅ Asset created: {asset.name} ({asset.symbol})")
        
    except SatoxError as e:
        print(f"⚠️  Asset Manager issue: {e}")
    
    # Initialize wallet manager
    print("\n5. Initializing Wallet Manager...")
    wallet_manager = WalletManager()
    
    try:
        await wallet_manager.initialize({})
        print("✅ Wallet Manager initialized")
        
        # Create a wallet
        wallet = await wallet_manager.create_wallet()
        print(f"✅ Wallet created: {wallet.address}")
        
    except SatoxError as e:
        print(f"⚠️  Wallet Manager issue: {e}")
    
    # Shutdown
    print("\n6. Shutting down...")
    try:
        await core_manager.shutdown()
        print("✅ SDK shutdown successfully")
    except SatoxError as e:
        print(f"⚠️  Shutdown issue: {e}")
    
    print("\n🎉 Example completed successfully!")

if __name__ == "__main__":
    asyncio.run(main()) 